### DISCUSSION TITLE
Permission problems trying to install PIM Community Edition [SOLVED]

### DISCUSSION THREAD

**Discussion by sebadamus** [akeneo/pim-community-dev#16165] [General]
Hi, I am new with Akeneo, trying to install CE version first to evaluate.  We are already working with Magento EE and would like to integrate with Akeneo.

I hope to be asking in the right place, if not please tell me.

I followed the manual steps here: [System Install Ubuntu 20.04](https://docs.akeneo.com/latest/install_pim/manual/system_requirements/system_install_ubuntu_2004.html) installed in a fresh new virtualbox, steps went good at least no errors appear.

Also checked and applied the steps here: [System Requeriments](https://docs.akeneo.com/latest/install_pim/manual/system_requirements/system_requirements.html)

The problem start trying to get this guide to work: [Installation CE](https://docs.akeneo.com/latest/install_pim/manual/installation_ce.html)

This is what I did:

1) I create a dir in my user home dir called pim, and run inside it: 
`composer create-project akeneo/pim-community-standard /srv/pim "5.0.*@stable"`

2) then composer create-project starts I accept with **Y** some "writes "allow-plugins" to composer.json) [y,n,d,?]" questions.
I attached the output composer create-project log:
[compose_create-project_log.txt](https://github.com/akeneo/pim-community-dev/files/7884341/compose_create-project_log.txt)
Seems to have ran ok without fatal errors I think... :-)

3) then I create the .env.local (as I read a not in .env that should use that field)
```
APP_ENV=prod
            APP_DATABASE_HOST=localhost
            APP_DATABASE_PORT=null
            APP_DATABASE_NAME=akeneo_pim
            APP_DATABASE_USER=akeneo_pim
            APP_DATABASE_PASSWORD=akeneo_pim
            APP_INDEX_HOSTS='localhost:9200'
```
5) run `NO_DOCKER=true make prod`

6) I copy the complete directory (hidden files also) to /var/www/html/ and chown to user/group www-data, and this as shown in the guide:
`sudo chmod ug+rw /var/www/html/var/cache /var/www/html/var/logs /var/www/html/public`

So it ended like this:
```
user1@ubuntuserver20:~/pim$ ll /var/www/html/
total 936
drwxr-xr-x 10 www-data www-data   4096 Jan 17 20:07 ./
drwxr-xr-x  3 root     root       4096 Jan 16 20:35 ../
drwxr-xr-x  2 www-data www-data   4096 Jan 17 20:07 .circleci/
-rw-r--r--  1 www-data www-data   1164 Jan 17 20:07 .env
-rw-r--r--  1 www-data www-data    770 Jan 17 20:07 .gitignore
-rw-r--r--  1 www-data www-data    143 Jan 17 20:07 LICENCE.txt
-rw-r--r--  1 www-data www-data   2912 Jan 17 20:07 Makefile
-rw-r--r--  1 www-data www-data   1768 Jan 17 20:07 README.md
drwxr-xr-x  2 www-data www-data   4096 Jan 17 20:07 bin/
-rw-r--r--  1 www-data www-data   1441 Jan 17 20:07 composer.json
-rw-r--r--  1 www-data www-data 414363 Jan 17 20:07 composer.lock
drwxr-xr-x  4 www-data www-data   4096 Jan 17 20:07 config/
drwxr-xr-x  3 www-data www-data   4096 Jan 17 20:07 docker/
-rw-r--r--  1 www-data www-data   3631 Jan 17 20:07 docker-compose.yml
-rw-r--r--  1 www-data www-data   1625 Jan 17 20:07 package.json
drwxr-xr-x  2 www-data www-data   4096 Jan 17 20:07 public/
drwxr-xr-x  2 www-data www-data   4096 Jan 17 20:07 src/
-rw-r--r--  1 www-data www-data    251 Jan 17 20:07 tsconfig.json
drwxr-xr-x 16 www-data www-data   4096 Jan 17 20:07 upgrades/
drwxr-xr-x 44 www-data www-data   4096 Jan 17 20:07 vendor/
-rw-r--r--  1 www-data www-data 462214 Jan 17 20:07 yarn.lock
```
7) Finally I tried to browse http://virtualboxIP and I get the list of files in documentroot... I dont understand if this is a official guide error (or misunderstanding) but there says you have to browse i.e. http://localhost:8080, but there is no port 8080 enabled in apache.

So I browse http://virtualboxIP/public and and finally there I get the login screen but with some images missing, if I enter admin/admin and click on login, it goes just to a blank apache "404 Not Found" default error.

Looking the logs I get this from **/var/log/apache2/access.log**
```
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /public/ HTTP/1.1" 401 6335 "http://10.10.0.125/" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /public/css/pim.css?c293690cd95a1c848ee49a9d8abe4b20 HTTP/1.1" 200 56719 "http://10.10.0.125/public/" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /bundles/pimui/images/logo_login.svg HTTP/1.1" 404 489 "http://10.10.0.125/public/" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /bundles/pimui/images/illustrations/login/Logo.svg HTTP/1.1" 404 490 "http://10.10.0.125/public/" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /bundles/pimui/fonts/Lato-Regular.ttf HTTP/1.1" 404 489 "http://10.10.0.125/public/css/pim.css?c293690cd95a1c848ee49a9d8abe4b20" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /bundles/pimui/images/logo_login.svg HTTP/1.1" 404 489 "http://10.10.0.125/public/" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /bundles/pimui/images/illustrations/login/Artboard.png HTTP/1.1" 404 489 "http://10.10.0.125/public/css/pim.css?c293690cd95a1c848ee49a9d8abe4b20" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
10.10.0.122 - - [19/Jan/2022:01:27:45 +0000] "GET /public/favicon.ico HTTP/1.1" 200 5263 "http://10.10.0.125/public/" "Mozilla/5.0 (X11; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0"
```

Hope you can help me to get Akeneo working?
(will try posting in StackOverFlow [akeneo] as suggested in "issues", have already dig every issue, but no luck)

Thanks,

(edited with updates)

**Comment by sebadamus** [comment]
Hi, I could solve the problem.

The [official manual](https://docs.akeneo.com/latest/install_pim/manual/system_requirements/system_install_ubuntu_2004.html) install guide tells you to create a new virtualhost on port *:80 called /etc/apache2/sites-available/**akeneo-pim.local.conf**
Then enable it.  If you do so, you wiil also have 000-default.conf enabled as it comes defaulted enabled with apache.
So, you have to replace the 000-default.conf contents shown [here](https://docs.akeneo.com/latest/install_pim/manual/system_requirements/system_install_ubuntu_2004.html#creating-the-virtual-host-file) in the official guide.
Then restart apache and it works, but in port 80 (not in por 8080 as told in the guide)
Also if you need to create a new user as the admin/admin is just for docker install, you can create user with this command ran in your documentroot akeneo folder.  Also you have to run it with the same user that runs apache or the documentroot akeneo directory permission (if not it will give cant write/access errors)

In my case apache and php7.4-fpm runs with www-data as the permissions in /var/www/html/ (my documentroot where akeneo CE is installed)

So to add user:
cd /var/www/html
sudo -u www-data php bin/console pim:user:create 

Hope it helps, will try to add the clean guide if it helps someone else.

**Comment by sebadamus** [comment]
(This is the guide modified, it worked for me)
How to install Akeneo PIM for testing/dev inUbuntu 20.04 LTS (my way)
------------------------------------------------------

Change localhost name (optional):

    sudo nano /etc/hosts
    	(add this or modify)
    	127.0.0.1	localhost
    	127.0.0.1	akeneo-pim.local

Install apache2 and modules:

    sudo apt install apache2 curl aspell php-apcu-bc php-pear php-apcu php7.4-fpm php7.4 libapache2-mod-php7.4 php7.4-cli php7.4-common php7.4-xml php7.4-mysql php7.4-gd php-imagick php7.4-tidy php7.4-xmlrpc php7.4-curl php7.4-mbstring php7.4-zip php7.4-soap php7.4-bcmath php7.4-intl php7.4-xsl php7.4-json zip unzip --install-recommends

(modify both php.ini)

    sudo nano /etc/php/7.4/cli/php.ini
    sudo nano /etc/php/7.4/fpm/php.ini
        memory_limit = 2G
        date.timezone = America/New_York
        apc.enable_cli = 1
    
    a2enmod rewrite proxy_fcgi
    
    sudo nano /etc/php/7.4/fpm/pool.d/www.conf    
        user = www-data
        group = www-data
        listen = /run/php/php7.4-fpm.sock
        listen.owner = www-data
        listen.group = www-data
        ## Optional: this can help to optimize if you have little memory in your virtual machine.
    	# pm = ondemand
    	# pm.max_children = 80
    	# pm.process_idle_timeout = 10s
    	# pm.max_requests = 200
       
    sudo service apache2 restart    
    sudo service php7.4-fpm restart
    
Add the Akeneo site to web server:    

    sudo mv /etc/apache2/sites-available/000-default.conf /etc/apache2/sites-available/000-default.conf.bak
    sudo nano /etc/apache2/sites-available/000-default.conf
        <VirtualHost *:80>
            ServerName akeneo-pim.local

            DocumentRoot /var/www/html/public
            
            <Directory /var/www/html/public>
                AllowOverride None
                Require all granted

                Options -MultiViews
                RewriteEngine On
                RewriteCond %{REQUEST_FILENAME} !-f
                RewriteRule ^(.*)$ index.php [QSA,L]
            </Directory>

            <Directory /var/www/html/public/bundles>
                RewriteEngine Off
            </Directory>

            <FilesMatch \.php$>
                SetHandler "proxy:unix:/run/php/php7.4-fpm.sock|fcgi://localhost/"
            </FilesMatch>

            SetEnvIf Authorization .+ HTTP_AUTHORIZATION=$0

            ErrorLog ${APACHE_LOG_DIR}/akeneo-pim_error.log
            LogLevel warn
            CustomLog ${APACHE_LOG_DIR}/akeneo-pim_access.log combined
        </VirtualHost>

        
Test if everything is working in the config (you might get some warning that will solve later)

       sudo apache2ctl configtest
       sudo service apache2 restart
    
Install Mysql, create db/user:

    sudo apt install mysql-server mysql-client --install-recommends
    	(Install phpmyadmin if you like):
    	sudo apt install phpmyadmin --install-recommends 
    
Create DB and user:       

    sudo mysql -u root -p
        CREATE DATABASE akeneo_pim;
        CREATE USER akeneo_pim@localhost IDENTIFIED WITH mysql_native_password BY 'akeneo-pim';
        GRANT ALL PRIVILEGES ON akeneo_pim.* TO akeneo_pim@localhost;
        FLUSH PRIVILEGES;
        EXIT
    
    Configure root with password (optional):
        GRANT ALL PRIVILEGES ON *.* TO root@localhost;
        FLUSH PRIVILEGES;
        EXIT;
        
    Add other users as root (optional):   
        CREATE USER 'youruser'@'localhost' IDENTIFIED BY 'CFKLDLNA'; 
        GRANT ALL PRIVILEGES ON *.* TO youruser@localhost;
        FLUSH PRIVILEGES;
        EXIT;
            
Install JDK and ElasticSearch:

    sudo apt install -y openjdk-11-jdk --install-recommends
    
    wget -qO - https://artifacts.elastic.co/GPG-KEY-elasticsearch | sudo apt-key add -
    sudo echo "deb https://artifacts.elastic.co/packages/7.x/apt stable main" | sudo tee /etc/apt/sources.list.d/elastic-7.x.list
    sudo apt update
    sudo apt install elasticsearch
    
Options to limit memory usage (test as you need)

        sudo nano /etc/elasticsearch/jvm.options.d/jvm.options
    	-Xms2g
    	-Xmx2g
    
    sudo systemctl daemon-reload
    sudo systemctl enable elasticsearch.service
    sudo service elasticsearch start

If Elasticsearch wont start at boot, give it little more timeout wait:
	
    sudo mkdir /etc/systemd/system/elasticsearch.service.d
    	echo -e "[Service]\nTimeoutStartSec=250" | sudo tee /etc/systemd/system/elasticsearch.service.d/startup-timeout.conf
    	sudo systemctl show elasticsearch | grep ^Timeout

Options for Akeneo:

    sudo sysctl -w vm.max_map_count=262144
    sudo nano /etc/sysctl.d/elasticsearch.conf
        vm.max_map_count=262144
    sudo service elasticsearch restart

Install NODE12:

    cd ~
    curl -sL https://deb.nodesource.com/setup_12.x -o nodesource_setup.sh
    sudo bash nodesource_setup.sh
    sudo apt install -y nodejs

Install YARN:

    cd ~
    wget https://dl.yarnpkg.com/debian/pubkey.gpg 
    sudo apt-key add pubkey.gpg 
    sudo nano /etc/apt/sources.list.d/yarn.list
        deb https://dl.yarnpkg.com/debian/ stable main
    sudo apt update && sudo apt install yarn    
    
Install Composer:

    cd ~
    	php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
    	php -r "if (hash_file('sha384', 'composer-setup.php') === '906a84df04cea2aa72f40b5f787e49f22d4c2f19492ac310e8cba5b96ac8b64115ac402c8cd292b8a03482574915d1a8') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
    	php composer-setup.php
    	php -r "unlink('composer-setup.php');"
    	sudo mv composer.phar /usr/local/bin/composer
    
Install Akeneo-PIM:

    sudo rm /var/www/html/*

(Must be an empty folder)
    
    cd /var/www/html
    sudo chown -R $USER:$USER /var/www/html
    php -d memory_limit=4G /usr/local/bin/composer create-project --prefer-dist akeneo/pim-community-standard . "5.0.*@stable"

(accept allow-plugins)      
        
Config .env as you need to match what you enter before:

    nano /var/www/html/.env.local 
        APP_ENV=prod
        APP_DATABASE_HOST=localhost
        APP_DATABASE_PORT=null
        APP_DATABASE_NAME=akeneo_pim
        APP_DATABASE_USER=akeneo_pim
        APP_DATABASE_PASSWORD=akeneo_pim
        APP_INDEX_HOSTS='localhost:9200'
           
Start installation in production mode:
(install this if you need: sudo apt make gcc)
        
    cd /var/www/html    
    NO_DOCKER=true make prod 
    sudo chown -R www-data:www-data /var/www/html
    sudo chmod ug+rw /var/www/html/var/cache /var/www/html/var/logs /var/www/html/public
        
Create an admin user:

    cd /var/www/html
    sudo -u www-data php bin/console pim:user:create
    User/Pass: youruser/CFKLDLNA
    
Cron job for Akeneo:

    sudo -u www-data /usr/bin/php /var/www/html/bin/console akeneo:batch:job-queue-consumer-daemon --env=prod -vvv >>/var/www/html/var/logs/daemon_logs.log 2>&1

**Reply by khuong97hd** [reply]
Can u help me ?, i can not run server after install akeneo pim done :(

**Reply by khuong97hd** [reply]
![image](https://user-images.githubusercontent.com/76488386/232988030-d866d581-ebe3-4c04-840f-52696911a15f.png)

**Reply by sebadamus** [reply]
Sorry friend, I  have done this loooong time ago.  But from your screenshot the problem is you dont have a PHP server working.  Thats why you see the initial php tag at the begining... its not interpreting the php code.  Try first to make the webserver to understand php, you might have done something wrong in the installation.
